src/arithmetic/arithmetic_expression.c

```diff
@@ -58,12 +58,26 @@ static AR_ExpNode *_AR_EXP_CloneOperand(AR_ExpNode *exp) {
 	return clone;
 }
 
+static AR_ExpNode *_AR_EXP_AllocateOpNode(const char *func_name, uint child_count) {
```

Comment:
This function does much more than simply allocating AR_ExpNode,
either rename accordingly or remove and duplicate these lines.